### PR TITLE
New Offline vocabulary

### PR DESCRIPTION
**Pull request by bruce-qin-dev**
New Offline vocabulary
New term ClientOnly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ This repository defines the following SAP vocabularies:
 - [PersonalData: Personal data / GDPR](vocabularies/PersonalData.md)
 - [Session: Sticky Sessions](vocabularies/Session.md)
 - [UI: Representing data in user interfaces](vocabularies/UI.md)
+- [Offline: Terms for Offline OData](vocabularies/Offline.md)
```
Comment (ralfhandl) [inline README.md:23]:
```suggestion
- [Offline: Terms for Offline OData](vocabularies/Offline.md) - _experimental_
```

Comment (ralfhandl) [reply README.md:23]: We mark a vocabulary as "experimental" until at least one of its terms is used in globally available software products.

File: examples/Offline.ClientOnly-sample.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      <EntityContainer Name="container">
+        <EntitySet Name="SalesOrder" EntityType="client.SalesOrderType" />
+      </EntityContainer>
+
+      <Annotations Target="client.SalesOrderType">
+        <Annotation Term="Offline.ClientOnly" />
+      </Annotations>
+
+      <Annotations Target="client.SalesOrder">
```
Comment (ralfhandl) [inline examples/Offline.ClientOnly-sample.xml:25]: If this annotation is targeting the entity set `SalesOrder`, the `Target` has to be
```suggestion
      <Annotations Target="client.container/SalesOrder">
```

Comment (ralfhandl) [inline examples/Offline.ClientOnly-sample.xml:25]: Why are you annotating both the entity type and the entity set? Shouldn't it suffice to annotate one of them?

File: vocabularies/Offline.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Alias="Core" Namespace="Org.OData.Core.V1" />
+  </edmx:Reference>
+  <edmx:DataServices>
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="com.sap.vocabularies.Offline.v1" Alias="Offline">
+      <Annotation Term="Core.Description">
+        <String>Terms for annotating Offline OData</String>
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:9]: What is "Offline OData"? Is this an SAP Product? If so, please add a link / short explanation.

File: vocabularies/Offline.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            <PropertyValue Property="href" String="https://sap.github.io/odata-vocabularies/vocabularies/Offline.json" />
+          </Record>
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href" String="https://github.com/sap/odata-vocabularies/blob/main/vocabularies/Offline.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="ClientOnly" Type="Core.Tag" DefaultValue="true" Nullable="false" AppliesTo="EntityType EntitySet EnumType ComplexType">
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:31]: Is this the full list of model elements that can be marked as `ClientOnly`? 

What about `TypeDefinition`? 

What about `NavigationProperty`? Can I have a client-only entity set and entity type, and a client-only navigation to it in an entity type that is "client-and-server"?

File: vocabularies/Offline.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        <Annotation Term="Core.Description" String="Target model is persisted on client device only"/>
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:32]:
```suggestion
        <Annotation Term="Core.Description" String="The annotated model element exists only on client device"/>
```

File: vocabularies/Offline.xml (hunk 4 above)
Comment (ralfhandl) [inline vocabularies/Offline.xml:31]:
```suggestion
      <Term Name="ClientOnly" Type="Core.Tag" DefaultValue="true" Nullable="false" AppliesTo="EntityType EntitySet EnumType ComplexType">
        <Annotation Term="Common.Experimental" />
```

Comment (ralfhandl) [reply vocabularies/Offline.xml:31]: Mark term as experimental until its first use in a globally available software product.

File: vocabularies/Offline.xml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Alias="Core" Namespace="Org.OData.Core.V1" />
+  </edmx:Reference>
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:5]:
```suggestion
  </edmx:Reference>
  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/Common.xml">
    <edmx:Include Namespace="com.sap.vocabularies.Common.v1" Alias="Common" />
  </edmx:Reference>
```

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: vocabularies/Offline.xml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href" String="https://github.com/sap/odata-vocabularies/blob/main/vocabularies/Offline.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="ClientOnly" Type="Core.Tag" DefaultValue="true" Nullable="false" AppliesTo="EntityType EntitySet EnumType ComplexType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="The annotated model element exists only on client device"/>
```
Comment (HeikoTheissen) [inline vocabularies/Offline.xml:36]: To clarify what "client-only" means, I suggest adding a statement like:
> The service as implemented on the server is described by the CSDL document with all elements removed that are tagged as `@Offline.ClientOnly`.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

File: examples/Offline.ClientOnly-sample.xml (hunk 2 above)
Comment (bruce-qin-dev) [reply examples/Offline.ClientOnly-sample.xml]: That's correct, thanks. Just annotate is enough. I just add some comments to the sample for clarification.

File: vocabularies/Offline.xml (hunk 3 above)
Comment (bruce-qin-dev) [reply vocabularies/Offline.xml:12]: Good idea,  a long description is added to introduce the product, thanks.

File: vocabularies/Offline.xml (hunk 4 above)
Comment (bruce-qin-dev) [reply vocabularies/Offline.xml]: TypeDefinition is added to the targets, thanks.
If entity set is local only,  then the NavigationPropertyBinding is implied to be local only.
The "client to server" relationship is currently prohibited.

File: vocabularies/Offline.xml (hunk 7 above)
Comment (bruce-qin-dev) [reply vocabularies/Offline.xml:44]: Nice point, added to the long description of the term, thanks.

File: vocabularies/Offline.xml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+         <Annotation Term="Core.LongDescription">
+           <String>Occasionally a customer will want to store additional “client-only”
+           entities in the same DB. The easiest way to accommodate this is for the client 
+           to extend the backend-returned $metadata (upon which the local persistence is 
+           based) and just to mark some entities as client-only so that the client won’t
+           attempt to upload any changes that are made to them locally.
+           The service as implemented on the server is described by the CSDL document with
+           all elements removed that are tagged as @Offline.ClientOnly.</String>
```
Comment (HeikoTheissen) [inline vocabularies/Offline.xml:52]:
```suggestion
           all elements removed that are tagged as `Offline.ClientOnly`.</String>
```
Just some formatting.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

File: vocabularies/Offline.xml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="The annotated model element exists only on client device"/>
+         <Annotation Term="Core.LongDescription">
+           <String>Occasionally a customer will want to store additional “client-only”
+           entities in the same DB. The easiest way to accommodate this is for the client 
+           to extend the backend-returned $metadata (upon which the local persistence is 
+           based) and just to mark some entities as client-only so that the client won’t
+           attempt to upload any changes that are made to them locally.
+           The service as implemented on the server is described by the CSDL document with
+           all elements removed that are tagged as `Offline.ClientOnly`.</String>
```
Comment (HeikoTheissen) [inline vocabularies/Offline.xml:52]:
```suggestion
           The service as implemented on the server is described by the CSDL document if
           all elements tagged as `Offline.ClientOnly` are removed together will all subsequently dangling references.</String>
```
Your example shows this addition is necessary: The `SalesOrderItem` entity set is not tagged, but must be removed together with the `SalesOrderItem` entity type.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

File: vocabularies/Offline.xml (hunk 9 above)
Comment (bruce-qin-dev) [reply vocabularies/Offline.xml]: That is true, thanks very much!

**Review by HeikoTheissen — DISMISSED** [review]

File: vocabularies/Offline.xml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    <edmx:Include Namespace="com.sap.vocabularies.Common.v1" Alias="Common" />
+  </edmx:Reference>
+  <edmx:DataServices>
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="com.sap.vocabularies.Offline.v1" Alias="Offline">
+      <Annotation Term="Core.Description">
+        <String>Terms for annotating Offline OData</String>
+      </Annotation>
+      <Annotation Term="Core.LongDescription">
+        <String>
+Offline OData is powerful solution for data sycnchronization between back-end and mobile devices.
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:16]:
```suggestion
Offline OData is a solution for data synchronization between back-ends and mobile devices.
```

File: vocabularies/Offline.xml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+It is built on top of in-house technology which has been designed and used in database synchronizing for over 20 years.
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:17]:
```suggestion
It is built on top of in-house technology which has been designed and used in database synchronization for over 20 years.
```

File: vocabularies/Offline.xml [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+Plese refer to the Offline OData product home for details:
+https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/features/offline/overview.html
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:19]:
```suggestion
Please refer to the [Offline OData product home](https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/features/offline/overview.html) for details.
```

File: vocabularies/Offline.xml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href" String="https://github.com/sap/odata-vocabularies/blob/main/vocabularies/Offline.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="ClientOnly" Type="Core.Tag" DefaultValue="true" Nullable="false" AppliesTo="EntityType EntitySet EnumType ComplexType TypeDefinition">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="The annotated model element exists only on client device"/>
```
Comment (ralfhandl) [inline vocabularies/Offline.xml:44]:
```suggestion
        <Annotation Term="Core.Description" String="The annotated model element exists only on client devices"/>
```

**Comment by ralfhandl** [comment]
Almost done, only a few typos and formatting issues left 😄

**Comment by bruce-qin-dev** [comment]
> Almost done, only a few typos and formatting issues left 😄

Thanks very  much for fixing the typos 👍 
Changes committed and MD/JSON files refreshed.

**Review by HeikoTheissen — DISMISSED** [review]

File: README.md [hunk 14]
```diff
@@ -20,6 +20,7 @@ This repository defines the following SAP vocabularies:
 - [PersonalData: Personal data / GDPR](vocabularies/PersonalData.md)
 - [Session: Sticky Sessions](vocabularies/Session.md)
 - [UI: Representing data in user interfaces](vocabularies/UI.md)
+- [Offline: Terms for Offline OData](vocabularies/Offline.md) - _experimental_
```
Comment (ralfhandl) [inline README.md:23]: Please move this new line up after the `ODM` line (line 18)

**Review by ralfhandl — CHANGES_REQUESTED** [review]

File: README.md (hunk 14 above)
Comment (bruce-qin-dev) [reply README.md]: Done, thanks a lot.

**Review by ralfhandl — APPROVED** [review]

**Review by HeikoTheissen — APPROVED** [review]

**Comment by bruce-qin-dev** [comment]
Hi @ralfhandl ,

In case no further change required, could you please help to merge the PR as we have no write permission, thanks a lot.

**Pull request merged**